File: applicatie/src/main/java/com/vdzon/administratie/contact/ContactResource.java
package com.vdzon.administratie.contact;

import com.vdzon.administratie.util.JsonUtil;
import spark.Spark;

import javax.inject.Inject;


public class ContactResource {

    @Inject
    public ContactResource(ContactService factuurService) {
        Spark.put("/rest/contact/", factuurService::putContact, JsonUtil.json());
        Spark.post("/rest/contact/", factuurService::putContact, JsonUtil.json());
        Spark.delete("/rest/contact/:uuid", factuurService::removeContact, JsonUtil.json());
    }


}

File: rest/src/main/java/com/vdzon/administratie/rest/contact/ContactResource.kt
package com.vdzon.administratie.rest.contact

import com.vdzon.administratie.util.JsonUtil
import spark.Route
import spark.Spark

import javax.inject.Inject


class ContactResource
@Inject
constructor(factuurService: ContactService) {
    init {
        Spark.put("/rest/contact/", Route { req, res -> factuurService.putContact(req, res) }, JsonUtil.json())
        Spark.post("/rest/contact/", Route { req, res -> factuurService.putContact(req, res) }, JsonUtil.json())
        Spark.delete("/rest/contact/:uuid", Route { req, res -> factuurService.removeContact(req, res) }, JsonUtil.json())
    }


}
